### PR TITLE
BDOG-391: Added called to get archived repositories, and filter out o…

### PR DESCRIPTION
**Pull request by JoPintoPaul**
…f team page

### CODE REVIEW

File: app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala [hunk 1]
```diff
@@ -98,6 +98,9 @@ class CatalogueController @Inject()(
   private lazy val jenkinsLinkName =
     configuration.getOptional[String]("teams-and-repositories.link-name.jenkins").getOrElse("jenkins")
 
+  private lazy val hideArchivedRepositoriesFromTeam: Boolean =
+    configuration.getOptional[Boolean]("team.hideArchivedRepositories").getOrElse(false)
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala:102]: It's preferable to put all default values in config - i.e. it's not optional. That way you don't need to look through code to see what the behaviour would be without it, and config-explorer would also display it (although it's not wired up for platapps).

File: app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala [hunk 2]
```diff
@@ -225,20 +228,24 @@ class CatalogueController @Inject()(
             userManagementConnector.getTeamDetails(teamName),
             leakDetectionService.repositoriesWithLeaks,
             serviceDependencyConnector.dependenciesForTeam(teamName),
-            serviceDependencyConnector.getCuratedSlugDependenciesForTeam(teamName, SlugInfoFlag.ForEnvironment(Environment.Production))).mapN {
-            (teamMembers, teamDetails, reposWithLeaks, masterTeamDependencies, prodDependencies) =>
+            serviceDependencyConnector.getCuratedSlugDependenciesForTeam(teamName, SlugInfoFlag.ForEnvironment(Environment.Production)),
+            if (hideArchivedRepositoriesFromTeam) teamsAndRepositoriesConnector.archivedRepositories else Future.successful(Nil)).mapN {
+            (teamMembers, teamDetails, reposWithLeaks, masterTeamDependencies, prodDependencies, archivedRepos) =>
+              val reposToHide: Seq[String] = archivedRepos.map(_.name)
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala:234]: could move the `.map(_.name)` above so the data is ready for `.mapN` i.e `.mapN {(.., reposToHide) => }`

File: app/uk/gov/hmrc/cataloguefrontend/connector/TeamsAndRepositoriesConnector.scala [hunk 3]
```diff
@@ -212,33 +209,31 @@ class TeamsAndRepositoriesConnector @Inject()(
...
 
   def digitalServiceInfo(digitalServiceName: String)(implicit hc: HeaderCarrier): Future[Option[DigitalService]] = {
     val url = teamsAndServicesBaseUrl + s"/api/digital-services/${encodePathParam(digitalServiceName)}"
     http.GET[Option[DigitalService]](url)
   }
 
   def allRepositories(implicit hc: HeaderCarrier): Future[Seq[RepositoryDisplayDetails]] =
+    http.GET[Seq[RepositoryDisplayDetails]](teamsAndServicesBaseUrl + s"/api/repositories")
+
+  def archivedRepositories(implicit hc: HeaderCarrier): Future[Seq[RepositoryDisplayDetails]] =
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/cataloguefrontend/connector/TeamsAndRepositoriesConnector.scala:232]: could create a func `def repositories(archived: Option[Boolean])` then can delegate `allRepositories` and `archivedRepositories` to it (or even just call directly passing the relevant `archived` param)

File: app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala (hunk 1 above)
Comment (colin-lamed) [reply app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala:102]: just seen that there is already a default in `application.conf`

File: app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala [hunk 4]
```diff
@@ -225,20 +228,24 @@ class CatalogueController @Inject()(
...
                   activityDates          = TeamActivityDates(teamInfo.firstActiveDate, teamInfo.lastActiveDate, teamInfo.firstServiceCreationDate),
                   errorOrTeamMembers     = convertToDisplayableTeamMembers(teamInfo.name, teamMembers),
                   errorOrTeamDetails     = teamDetails,
                   umpMyTeamsUrl          = umpMyTeamsPageUrl(teamInfo.name),
                   leaksFoundForTeam      = leakDetectionService.teamHasLeaks(teamInfo, reposWithLeaks),
                   hasLeaks               = leakDetectionService.hasLeaks(reposWithLeaks),
-                  masterTeamDependencies = masterTeamDependencies,
-                  prodDependencies       = prodDependencies
+                  masterTeamDependencies = masterTeamDependencies.filterNot(dependencies => reposToHide.contains(dependencies.repositoryName)),
+                  prodDependencies       = prodDependencies.filterNot { case (repo, _) => reposToHide.contains(repo) }
```
Comment (colin-lamed) [inline app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala:248]: maybe don't need to filter `prodDependencies` - if it's in production we would want to know about it

File: app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala (hunk 1 above)
Comment (JoPintoPaul) [reply app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala]: @colin-lamed Addressed this.

File: app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala (hunk 2 above)
Comment (JoPintoPaul) [reply app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala]: Done.

File: app/uk/gov/hmrc/cataloguefrontend/connector/TeamsAndRepositoriesConnector.scala (hunk 3 above)
Comment (JoPintoPaul) [reply app/uk/gov/hmrc/cataloguefrontend/connector/TeamsAndRepositoriesConnector.scala:232]: Done. I prefer to keep it as two functions externally as I think it is more self-documenting that way.

File: app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala (hunk 4 above)
Comment (JoPintoPaul) [reply app/uk/gov/hmrc/cataloguefrontend/CatalogueController.scala]: Good point. Done.

**Pull request merged**